xcode/Subconscious/SubconsciousTests/Tests_SphereFS.swift

```diff
@@ -0,0 +1,496 @@
+            sphereStoragePath: sphereStoragePath,
+            gatewayURL: "http://fake-gateway.fake"
+        )
+        
+        let sphereReceipt = try noosphere.createSphere(ownerKeyName: "bob")
+        
+        let sphere = try SphereFS(
+            noosphere: noosphere,
+            identity: sphereReceipt.identity
+        )
+        
+        // Should fail
+        _ = try? sphere.sync()
+        
+        try sphere.write(
+            slug: "foo",
+            contentType: "text/subtext",
+            body: "Test".toData(encoding: .utf8)!
+        )
+        
+        try sphere.save()
+        
+        let foo = try sphere.read(slashlink: "/foo")
+        
+        // Should fail
+        _ = try? sphere.sync()
+        
+        XCTAssertEqual(
+            foo.body.toString(),
+            "Test",
+            "Read current version"
+        )
+    }
+    
+    func testWritesThenCloseThenReopenWithScopes() throws {
+        let base = UUID()
+        
+        let globalStoragePath = try createTmpDir(path: "\(base)/noosphere")
+            .path()
+        
+        let sphereStoragePath = try createTmpDir(path: "\(base)/sphere")
+            .path()
+        
+        let sphereIdentity: String
+        let startVersion: String
+        let endVersion: String
+        do {
+            let noosphere = try Noosphere(
+                globalStoragePath: globalStoragePath,
+                sphereStoragePath: sphereStoragePath
+            )
+            
+            let sphereReceipt = try noosphere.createSphere(ownerKeyName: "bob")
+            
+            sphereIdentity = sphereReceipt.identity
+            
+            let sphere = try SphereFS(
+                noosphere: noosphere,
+                identity: sphereReceipt.identity
+            )
+            
+            startVersion = try sphere.version()
+            
+            let body = try "Test content".toData().unwrap()
+            let contentType = "text/subtext"
+            try sphere.write(slug: "a", contentType: contentType, body: body)
+            try sphere.write(slug: "b", contentType: contentType, body: body)
+            try sphere.write(slug: "c", contentType: contentType, body: body)
+            let version = try sphere.save()
+            endVersion = try sphere.version()
+            XCTAssertEqual(version, endVersion)
+            XCTAssertNotEqual(startVersion, endVersion)
+        }
+        
+        do {
+            let noosphere = try Noosphere(
+                globalStoragePath: globalStoragePath,
+                sphereStoragePath: sphereStoragePath
+            )
+            
+            let sphere = try SphereFS(
+                noosphere: noosphere,
+                identity: sphereIdentity
+            )
+            let version = try sphere.version()
+            
+            XCTAssertEqual(version, endVersion)
+        }
+    }
+    
+    func testWritesThenCloseThenReopenWithVars() throws {
```

Comment:
This test fails.

Note I overwrite the old noosphere variable with a new one in the same scope. It is therefore possible that the Noosphere deinit does not get called until the scope exits. deinit is where we deallocate the Noosphere pointer.